src/requests.js

```diff
@@ -64,6 +64,44 @@ export function updateRequest(path, { data, permissions }, options = {}) {
 /**
  * @private
  */
+export function jsonPatchPermissionsRequest(
+  path,
+  permissions,
+  opType,
+  options = {}
+) {
+  const {
+    headers,
+    safe,
+    last_modified,
+  } = { ...requestDefaults, ...options };
+
+  const ops = [];
+
+  for (const type in permissions) {
+    for (const principal in permissions[type]) {
+      ops.push({
+        op: opType,
+        path: "/permissions/" + type + "/" + permissions[type][principal],
+      });
+    }
+  }
+
+  return {
+    method: "PATCH",
+    path,
+    headers: {
+      "Content-Type": "application/json-patch+json",
```

Comment:
I'd probably put this one at the last position so we're sure it's always applied. Here if `headers` contains a `Content-Type`, the one we need is overridden.